main.go

```diff
@@ -152,21 +175,64 @@ func main() {
 				generateCmd.Env = append(generateCmd.Env,
 					fmt.Sprintf("VIRTINK_CONTROL_PLANE_MACHINE_ROOTFS_CDI_IMAGE=%s", controlPlaneMachineRootfsImage),
 					fmt.Sprintf("VIRTINK_WORKER_MACHINE_ROOTFS_CDI_IMAGE=%s", workerMachineRootfsImage),
-					fmt.Sprintf("VIRTINK_NODE_ADDRESSES=[%v]", strings.Join(machineAddresses, ",")),
+					fmt.Sprintf("VIRTINK_IP_POOL_NAME=%s", args[0]),
 				)
 
+				if err := runCommand(exec.Command("kubectl", "delete", "ippool.ipam.metal3.io", args[0], "--namespace", targetNamespace, "--wait", "--ignore-not-found")); err != nil {
+					return fmt.Errorf("delete IPPool: %s", err)
+				}
+
+				ipPool := ipamv1.IPPool{
```

Comment:
I don't think it's a good idea to use the CRD object as data to execute the template. If we have to use the CRD object anyway, we can just marshal it to yaml. The point of using yaml template is to avoid ref the CRD object in the code.